### ISSUE TITLE
Can't run Authelia on Ubuntu 20.04 because of glibc version mismatch

### ISSUE THREAD

**Issue by ntimo** [authelia/authelia#6924]
### Version

v4.38.3

### Deployment Method

Bare-metal

### Reverse Proxy

HAProxy

### Reverse Proxy Version

2.8

### Description

When trying to run the authelia binary from github on Ubuntu 20.04 the binary fails to start because it was compiled with a newer glibc version "authelia: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by authelia)". This is quite bad since it would force users to upgrade their baremetall install to a newer Ubuntu version e.g. 22.04 just to get the new glibc version.

### Reproduction

Try to run the Authelia binary on Ubuntu 20.04 and notice that it does not start

### Expectations

Authelia should work on 20.04

### Configuration (Authelia)

```yaml
Not relevant
```

### Build Information

```shell
Not possible since the binary fails to start
```

### Logs (Authelia)

```shell
authelia: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by authelia)
```

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)

- [X] This is a bug report and not a support request

- [X] I have read the security policy and this bug report is not a security issue or security related issue

- [X] I have either included the complete configuration file or I am sure it's unrelated to the configuration

- [X] I have either included the complete debug / trace logs or the output of the build-info command if the logs are not relevant

- [X] I have provided all of the required information in full with the only alteration being reasonable sanitization in accordance with the [Troubleshooting Sanitization](https://www.authelia.com/r/sanitize) reference guide

- [X] I have checked for related proxy or application logs and included them if available

- [X] I have checked for related issues and checked the documentation

**Comment by james-d-elliott** [comment]
This is really a feature request to support a dated OS. As far as I can tell there is no real criteria here as to what OSes we support and for how long.

This issue occurs due the the fact we build position independent execuables with full relocation read-only to prevent return orientated programming and various other on-system memory exploits making it a requirement to build dynamically.

**Comment by nightah** [comment]
We have just updated our build tooling to support what is considered `oldstable` in Debian and Ubuntu.

This will be built with glibc 2.31 and will land in Authelia v4.38.4. Hopefully to be released tonight.

I'll leave this open for tracking purposes and we can tag v4.38.4 to close this issue off. Please subscribe to see when this lands.

**Comment by nightah** [comment]
This should be resolved by #6938 (1ea41b80d3a44e30d299acf04e6e18afd3836455 ).
